akka-stream-tests/src/test/scala/akka/stream/impl/fusing/IteratorInterpreterSpec.scala

```diff
@@ -47,29 +48,21 @@ class IteratorInterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
       a[NoSuchElementException] should be thrownBy { itr.next() }
     }
 
-    "throw exceptions when chain fails" in {
-      val stage = new PushStage[Int, Int] {
-        override def onPush(elem: Int, ctx: Context[Int]): SyncDirective = {
-          if (elem == 2) ctx.fail(new ArithmeticException())
-          else ctx.push(elem)
-        }
-      }
-      val itr = new IteratorInterpreter[Int, Int](List(1, 2, 3).iterator, Seq(stage.toGS)).iterator
-
-      itr.next() should be(1)
-      itr.hasNext should be(true)
-      a[ArithmeticException] should be thrownBy { itr.next() }
-      itr.hasNext should be(false)
-    }
-
     "throw exceptions when op in chain throws" in {
-      val stage = new PushStage[Int, Int] {
-        override def onPush(elem: Int, ctx: Context[Int]): SyncDirective = {
-          if (elem == 2) throw new ArithmeticException()
-          else ctx.push(elem)
-        }
+      val stage = new SimpleLinearGraphStage[Int] {
+        override def createLogic(attributes: Attributes): GraphStageLogic =
+          new GraphStageLogic(shape) with InHandler with OutHandler {
+            override def onPush(): Unit = {
+              val elem = grab(in)
+              if (elem == 2) throw new ArithmeticException()
+              else push(out, elem)
+            }
+            override def onPull(): Unit = pull(in)
+
+            setHandlers(in, out, this)
+          }
```

Comment:
couldn't this just be replaced by a `Flow[Int].map(n => if (n == 2) throw new ArithmeticException() else n)`?